js-foundations.js

```diff
@@ -33,18 +62,73 @@ var testString3 = "sandiego"
 
 // 4a. Write a function called alphabetSoup that takes one argument and returns a string with all the letters in alphabetical order. Use each of the varibales as test cases. Expected output: "aelnr" "aacdemy" "adeginos"
 
+const inOrder = (str) => {
+  return str.split('').sort().join('')
+}
+console.log(inOrder(testString1));
+console.log(inOrder(testString2));
+console.log(inOrder(testString3));
 
 
 // 4b. Write a function that takes in all three variables and returns a string with all the letters in alphabetical order. Expected output: "aaaacddeeegilmnnorsy"
 
 
+const inOrder2 = (str1, str2, str3) => {
+  let newStr = str1.concat(str2, str3)
+
+  let orderedStr = newStr.split('').sort().join('')
+
+  return orderedStr
+}
+console.log(inOrder2(testString1, testString2, testString3));
 
 // ------------------------------------- Consider the variables:
 var amounts = [9, 1, 8, 16, 5, 1, 42]
 var animals = ["ducks", "elephants", "pangolins", "zebras", "giraffes", "penguins", "llamas"]
 
 // 5a. Write a function that takes the two variables and returns a combined string. Expected output: "9 ducks 1 elephants 8 pangolins 16 zebras 5 giraffes 1 penguins 42 llamas"
 
-
+const animalCount = (arr1, arr2) => {
+  let newArr = arr1.map((x,y) => {
+    return x +  ' ' + arr2[y] 
+  }
+ )
+ return newArr.join(' ')
+}
+console.log(animalCount(amounts, animals))
+
+// ----------------------------------------------------------------------------------------- //
+
+
+const theWorstFunction = (arr1, arr2) => {
+    const maybeEvenWorseFunction = (arr1, arr2) => {
+      console.log(`${arr1} ${arr2}`);
+    }
+    maybeEvenWorseFunction(amounts[0],animals[0])
+    maybeEvenWorseFunction(amounts[1],animals[0])
+    maybeEvenWorseFunction(amounts[2],animals[0])
+    maybeEvenWorseFunction(amounts[3],animals[0])
+    maybeEvenWorseFunction(amounts[4],animals[0])
+    maybeEvenWorseFunction(amounts[5],animals[0])
+    maybeEvenWorseFunction(amounts[6],animals[0])
+  }
+theWorstFunction()
 
 // 5b. STRETCH: Create a function that combines the two variables and updates the animal to be singular if the animal's corresponding number is 1. Expected output: "9 ducks 1 elephant 8 pangolins 16 zebras 5 giraffes 1 penguin 42 llamas"
+
+
+const animalCount2 = (arr1, arr2) => {
+
+  return arr1.map((x, y) => {
+    if (x <= 1) {
+      arr2[y] = arr2[y].slice(0, arr2[y].length -1)
+      let final = x + ' ' + arr2[y]
+      return final;
+    } else {
+      let final =  x + ' ' + arr2[y]
+      return final
+    }
+    }
+  )
+}
+console.log(animalCount2(amounts, animals))
```

Comment:
Looks great, maybe add a join so you are returning a string rather than an array.